File: api/src/main/java/com/mengyunzhi/springBootStudy/entity/Term.java
package com.mengyunzhi.springBootStudy.entity;

import com.fasterxml.jackson.annotation.JsonFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
public class Term {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    private School school;

    private String name;
    private Long state;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", timezone = "GMT+8")
    private Date startTime;

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd", timezone = "GMT+8")
    private Date endTime;

    public Term() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public School getSchool() {
        return school;
    }

    public void setSchool(School school) {
        this.school = school;
    }

    public Long getState() {
        return state;
    }

    public void setState(Long state) {
        this.state = state;
    }

    public Date getStartTime() { return startTime; }

    public void setStartTime(Date startTime) { this.startTime = startTime; }

    public Date getEndTime() { return endTime; }

    public void setEndTime(Date endTime) { this.endTime = endTime; }
}

File: web-app/src/app/norm/entity/Term.ts
import {School} from './School';

/**
 * 班级实体
 */
export class Term {
  id: number | undefined;
  name: string | undefined;
  school: School | undefined;
  startTime: Date;
  endTime: Date;

  /**
   * 构造函数
   * @param id id
   * @param name 名称
   * @param school 学校
   */
  constructor(id: number | undefined, name: string | undefined, school: School | undefined, startTime: Date, endTime: Date) {
    this.id = id;
    this.name = name;
    this.school = school;
    this.startTime = startTime;
    this.endTime = endTime;
  }
}
